### ISSUE TITLE
Wrong operator in composer.json

### ISSUE THREAD

**Issue by Christoph-Harms** [aws/aws-sdk-php#1205]
According to [the composer documentation](https://getcomposer.org/doc/articles/versions.md#range), a **double pipe** must be used to express a logical OR in version constraints.

Current version constraint for guzzle in the composer.json is

`"guzzlehttp/guzzle": "^5.3.1|^6.2.1"`

(With that single pipe, the version constraint will never resolve to ^6.2.1)

**Comment by imshashank** [comment]
Thanks for bringing this to our notice, please give us some time to repro the issue on our end and verify.

**Comment by Christoph-Harms** [comment]
To add to the discussion, take a look at jarektkaczyk/eloquence#155 and the Issue mentioning it.

**Comment by mtdowling** [comment]
Aren't they both valid? They would have to be for Composer to even work.

What version of Composer are you using? I thought "|" was added after Composer 1.0 launched? I'm asking for clarification here, because for us to change this, we'll need to communicate a minimum version requirement of Composer to our customers (both external and AWS internal). We'll also need to ensure this wont affect Amazon Elastic Beanstalk as they have Composer integration.

**Comment by Christoph-Harms** [comment]
Welp, now I'm not sure anymore... The documentation clearly states that a logical OR is expressed by a double pipe and makes no mention of a single pipe. I'll set up a test project as soon as I have the time (probably this weekend).

**Comment by kstich** [comment]
[Extracts from older versions of the docs](http://qpleple.com/understand-composer-versions/) show that, at least at some point, a single pipe was valid. It's likely supported in an undocumented fashion since it didn't seem to break when I ran a `composer install` earlier today.

I got 6.2.3 for guzzle, which is valid under the `^6.2.1` portion of the flag.
>   - Installing guzzlehttp/guzzle (6.2.3): Downloading (100%)
